repo-cleaner

```diff
@@ -1,81 +1,151 @@
+#/
+#/   * Lists repos under user account `IAmHughes` that are available to be deleted.
+#/   $ bash repo-cleaner -u=IAmHughes
+#/
+#/   * Deletes repos under org "TheBeardedTom" that are named "Test_Repo_*"
+#/     For example, Test_Repo_, Test_Repo_qa, Test_Repo_234, etc.
 #/   $ bash repo-cleaner -o=TheBeardedTom -s=Test_Repo_
 #/
-#/   * Deletes repos under user "IAmHughes" that are named "MyRepo*" - For example, MyRepo, MyRepo1, MyRepo99, MyRepoQA, etc.
+#/   * Deletes repos under user "IAmHughes" that are named "MyRepo*"
+#/     For example, MyRepo, MyRepo1, MyRepo99, MyRepoQA, etc.
 #/   $ bash repo-cleaner -u=IAmHughes -s=MyRepo
 #/
 #/ API DOCUMENTATION:
 #/ All documentation can be found at https://developer.github.com/v3/
 
-while getopts ":v" opt; do
-  case ${opt} in
-    v ) # process option a
-      verbose=true
-      shift
-      ;;
-    \? ) echo "Usage: cmd [-v]"
-      ;;
+# Parse options
+for i in "$@"
+do
+  case $i in
+    -o=*|--org=*)
+    ORG="${i#*=}"
+    shift
+    ;;
+    -u=*|--user=*)
+    USERNAME="${i#*=}"
+    shift
+    ;;
+    -s=*|--string=*)
+    STRING_MATCH="${i#*=}"
+    shift
+    ;;
+    *)
+    # unknown option
+    ;;
   esac
 done
 
-OWNER=$1
-STRING_MATCH=$2
+API_ROOT="https://api.github.com"
+
+# Verify Inputs
+# Must set either org or user.
+if [[ -z ${ORG} ]] && [[ -z ${USERNAME} ]]; then
+    echo "No user or org was given. Please use either -u=<username> or -o=<org_name>"
+    echo "  Ex: bash repo-cleaner -u=IAmHughes -s=MyRepo"
+    echo ""
+    echo "Exiting script with no changes."
+    echo ""
+    exit 1
+fi
 
+# Cannot set BOTH org and user.
+if [[ ${ORG} ]] && [[ ${USERNAME} ]]; then
+  echo "Cannot set both an org and user. You must use either -u=<username> or -o=<org_name>, "
+  echo "not both."
+  echo "  Ex: bash repo-cleaner -u=IAmHughes -s=MyRepo"
+  echo ""
+  echo "Exiting script with no changes."
+  echo ""
+  exit 1
+fi
+
+# String Match null check, if it is will list repos for org or user.
 if [[ -z ${STRING_MATCH} ]]; then
- echo “No comparison string has been set to delete repos. Listing repos instead.”
+ echo "No comparison string has been set to delete repos. Listing repos instead."
+ echo ""
 fi
 
-RESPONSE=“$(curl --url https://api.github.com/orgs/${OWNER}/repos \
+# Set OWNER as either user or org and sets api endpoint
+if [[ ${USERNAME} ]]; then
+  OWNER=${USERNAME}
+  API_ENDPOINT=${API_ROOT}"/users"
+elif [[ ${ORG} ]]; then
+  OWNER=${ORG}
+  API_ENDPOINT=${API_ROOT}"/orgs"
+fi
+
+# Get list of repos from GitHub API for ${OWNER}
+RESPONSE=“$(curl --url ${API_ENDPOINT}/${OWNER}/repos \
  -s \
- --header “authorization: Bearer ${GITHUB_API}“)”
+ --header “authorization: Bearer ${GITHUB_TOKEN}“)”
 
-for repo in $(echo “${RESPONSE}” | jq -r ‘.[] | @base64'); do
-   _jq() {
+# Loop through list of repos. If ${STRING_MATCH} has a pattern, delete those
+# repos, otherwise just display the list of repos returned.
+for repo in $(echo ${RESPONSE} | jq -r '.[] | @base64');
```

Comment:
Got it! It works if you set it to:

`for repo in $(echo "${RESPONSE}" | jq -r '.[] | @base64');`

The quotes matter, apparently :)

Reply:
Added quotes in latest commit, didn't fix the error 😢